src/D2L.CodeStyle.SpecTests/SpecTestFixtureBase.cs

```diff
@@ -327,11 +328,49 @@ ImmutableArray<MetadataReference> metadataReferences
 			return solution.Projects.First().GetCompilationAsync();
 		}
 
-		private sealed record PrettyDiagnostic(
-			string Id,
-			LinePositionSpan LinePosition,
-			string Message
-		) {
+		private sealed class PrettyDiagnostic {
+
+			internal readonly record struct Line(
+				int LineNumber,
+				string Text
+			);
+
+			private PrettyDiagnostic(
+				string id,
+				LinePositionSpan linePosition,
+				string message,
+				ImmutableArray<Line> lines
+			) {
+				Id = id;
+				LinePosition = linePosition;
+				Message = message;
+				Lines = lines;
+			}
+
+			public string Id { get; }
+			public LinePositionSpan LinePosition { get; }
+			public string Message { get; }
+			public ImmutableArray<Line> Lines { get; }
+
+			public override bool Equals( object obj ) {
+				if( this is null ) {
+					return obj is null;
+				}
+
+				if( obj is not PrettyDiagnostic other ) {
+					return false;
+				}
+
+				return Id.Equals( other.Id )
+					&& LinePosition.Equals( other.LinePosition )
+					&& Message.Equals( other.Message, StringComparison.Ordinal );
+			}
+
+			public override int GetHashCode() => HashCode.Combine(
+				Id,
+				LinePosition,
+				Message
+			);

```

Comment:
Value equality falls apart with the addition of the immutable array.

Ignoring it in equals and gethashcode